rust/src/memory_pool.rs

```diff
@@ -0,0 +1,54 @@
+use libc;
+use std::sync::Arc;
+use std::mem;
+
+use super::error::ArrowError;
+use super::error::Result;
+
+const ALIGNMENT: usize = 64;
+
+pub trait MemoryPool {
+  fn allocate_aligned(&self, size: usize) -> Result<*mut u8>;
```

Comment:
This should also get a `Reallocate` function as we have in C++ https://github.com/apache/arrow/blob/6633cc94c84f9ecc516f03175da56361cf4bd67f/cpp/src/arrow/memory_pool.h#L77

When using jemalloc, this was one of the major performance improvements that we could avoid 50% of all reallocations.

For starters, it will be ok to always allocate a new block of memory.

Reply:
Hi, xhochy:
Thanks for the review and I've added an implementation for that method.